docs/assignment2.md

```diff
@@ -1 +1,322 @@
+* _loadSnippets(folder:File): void_
+    * uses JsonIO to read all snippets in the `pathToSnippoDir` folder.
+* _listSnippets(snippetsToList:Map<Integer, Snippet>): String_
+    * given an integer-snippet map, prints the list of snippets.
+* _listAll(): String_
+    * same as listSnippets but for all snippets managed.
+* _create(title:String, content:String, language:String, tags :String[*]): Integer_
+    * creates a snippet object with the given arguments.
+* _read(id:Integer): Snippet_
+    * returns the snippet with the provided id.
+* _delete(id:Integer): void_
+    * deletes the snippet with the provided id.
+* _edit(id:Integer): void_
+    * calls the snippets edit method.
+* _filter(wordToContain:String, tags:String[*], language:String): HashMap<Integer, Snippet>_
+    * filters the snippets managed and returns an integer-snippet map.
+* _getNextId(): Int_
+    * gets the next available id for a new snippet.
+* _isValidId(id:Integer): Boolean_
+    * checks if an integer is a valid id for an existing snippet.
+* _generatePathToSnippetJson(snippetId:Integer): String_
+    * returns the relative path for the requested snippet (by id).
+* _getSnippets(): HashMap<Integer; Snippet>_
+    * returns all the managed snippets as an integer-snippet hashmap.
+#### Associations
+* _SnippetManager < Main_
+    *  On the start of the program the main class constructs a snippetmanager class which maintains all the operations on the snippets.
+* _SnippetManager > Snippet_
+    * described
+* _SnippetManager > JsonIO_
+    * described
+    
+### class: JsonIO
+This class takes care of the conversion between string type and Json type using the GSON library.
+#### Attributes
+* _g: Gson_
+    * contains the Gson object
+#### Operations
+* _JsonIO(): void_
+    * instantiatest a Gson object.
+* _getInstance(): JsonIO_
+    * returns the singleton json object.
+* _writeToJson(pathToJson:String, object :Object): void_
+    * serializes an object.
+* _loadFromJson(pathToJson:String): Snippet_
+    * deserializes a Snippet.
+* _onException(e:Exception): void_
+    * handles exceptions like file not found or bad data.
+#### Associations
+* _JsonIO < Main_
+    * already described
+* _JsonIO < Snippet_
+    * The snippet class uses the JsonIO class to convert snippet objects to json, to store snippet objects.
+* _JsonIO < SnippetManager_
+    * The snippet manager class uses the JsonIO class to convert json to objects, to retrieve snippet objects.
+
+### class: Main
+ 
+#### Attributes
+* _snippetManager: SnippetManager_
+    * the snippet manager used by the current instance of the app.
+* _cliUI: CliUI_
+    * the cliUI used by the current instance of the app.
+#### Operations
+* _main(args:String[*]): void_
+    * Instantiates SnippetManager, CliUI and JsonIO. After this, CliUI handles the flow of the app.
+#### Associations
+* _Main > JsonIO_
+    * instantiates.
+* _Main > SnippetManager_
+    * instantiates.
+* _Main > CliUI_
+    * instantiates.
+    
```

Comment:
-Snippet class looks ok
-The editor class looks like it produces one instance and logically the multiplicity for it should be one, whereas there are many snippets so the multiplicity should be *
-Multiplicity for Snippet is not that visible, you could've had the association name from Editor directed to Snippet being just 'edits'
-Lacking multiplicity on the Snippet-Snippet Manager composition- should be one.
-You write the Editor class is in directed relation with the SnippetManager class but such a thing is not shown in the class diagram?
-CliUI multiplicity should be 1 if it's one overarching interface manager
-Good use of information hiding
-Ok separation between UI and functionality